packages/wrangler/src/logger.ts

```diff
@@ -118,6 +107,17 @@ export class Logger {
 		Logger.#afterLogHook?.();
 	}
 
+	static warnOnceHistory = new Set();
```

Comment:
Why is this static? It feels like each logger instance should have it's own warnOne set

Reply:
If a subclass wants a new set, it can override its own static property i.e. `static warnOnceHistory = new Set()` on the class

I guess if it was an instance property, the inverse could also be said: If a subclass wants to share the set, it can override its own instance property i.e. `this.warnOnceHistory = logger.warnOnceHistory` in the constructor

The former seems simpler but not tied to either tbh

Reply:
Ah sorry you said instance not subclass

The logger is currently a singleton so I'm more concerned with subclasses (we do this with Miniflare's Log class to intercept method calls, where we would expect this kind of property to be shared).

From this feature's perspective, I'd want these warnings to be logged once per invocation of wrangler not per instance of the logger

Reply:
The logger is used as a singleton in Wrangler, but there's no inherit reason it has to be, and in various tests we construct multiple instances. It seems to me that it would be kinda strange for those instances to share this warnOnce set.

In terms of subclasses, maybe we're coming at this from different angles, but I would be really surprised if a method call on an instance of the parent class affected method calls on an instance of a subclass.